### PR TITLE
fix(refinement-list): re-apply focus on facet checkbox after render

### PR DESCRIPTION
**Pull request by dhayab**
**Summary**

In InstantSearch.js, focus is lost when toggling a facet value in a refinement list. This is because the component is destroyed and re-created during re-render, and [Preact does not explicitly refocus the newly created component](https://github.com/preactjs/preact/issues/3242) (whereas React does it).

This PR adds code that stores the last refined value and applies focus to its DOM element on re-render.

CR-6976

### CODE REVIEW

File: packages/instantsearch.js/src/components/RefinementList/RefinementList.tsx [hunk 1]
```diff
@@ -270,6 +274,20 @@ class RefinementList<TTemplates extends Templates> extends Component<
     }
   }
 
+  /**
+   * This sets focus on the last refined input element after a render
+   * because Preact does not perform it automatically.
+   * @see https://github.com/preactjs/preact/issues/3242
+   */
+  public componentDidUpdate() {
```
Comment (Haroenv) [inline packages/instantsearch.js/src/components/RefinementList/RefinementList.tsx:282]: too bad to have this fix in the non-unified components 🤔

Comment (dhayab) [reply packages/instantsearch.js/src/components/RefinementList/RefinementList.tsx:282]: Yes, we should brace for a few headaches when we'll migrate all of this into the unified components package 😰.

File: packages/instantsearch.js/src/components/RefinementList/__tests__/RefinementList-test.tsx [hunk 2]
```diff
@@ -692,5 +693,57 @@ describe('RefinementList', () => {
 
       expect(toggleRefinement).toHaveBeenCalledTimes(0);
     });
+
+    it('should keep focus on toggled input between re-renders', () => {
```
Comment (Haroenv) [inline packages/instantsearch.js/src/components/RefinementList/__tests__/RefinementList-test.tsx:697]: this test should be in cts as well so it won't be reintroduced when we migrate components.

**Review by Haroenv — COMMENTED** [review]
fix looks like a good workaround for now. 

- is this happening anywhere else? What makes refinement list special that it doesn't need to be fixed somewhere else? no form elements in other components?
- test should be in CTS, maybe for all components?

**Review by Haroenv — APPROVED** [review]

**Comment by Haroenv** [comment]
I guess it won't matter if it happens anywhere else? focus can still be lost on a link right?

**Pull request merged**